### DISCUSSION TITLE
Multiple Stream Viewers

### DISCUSSION THREAD

**Discussion by p-moran-gd** [easytarget/esp32-cam-webserver#274] [Common Issues]
G'day,

Just wondering if there is anyway to have multiple people view the stream at once. As it currently stands, only one person/view at a time is supported. Once one closes the next person who asked for the stream gets it and so on.
Anyway to have the stream loaded, and simply viewed, so multiple people can see the same thing?

**Comment by daschr** [comment]
You may take a look at the multi-stream branch: https://github.com/easytarget/esp32-cam-webserver/tree/multi-stream
I've implemented the feature months ago, but unfortunaly, it's still not merged into the main branch.

You can just switch to the branch and build it including your `myconfig.h`.

**Reply by p-moran-gd** [reply]
Thanks so much for this.

**Reply by Zigouille22** [reply]
Hello, Is the feature active in the main branch ? i need multiple viewers for a bird cam project.
if not how can i do ?

**Reply by daschr** [reply]
Just clone/checkout the multi-stream branch and configure it like the main one.

F.e.: `git clone https://github.com/easytarget/esp32-cam-webserver.git --branch multi-stream`

**Reply by Zigouille22** [reply]
i clone multi-stream branch, and config the myconfig.h but when i compile the program i have these errors 
httpd_socket_send' was not declared in this scope
and error: 'CAMERA_GRAB_LATEST' was not declared in this scope
do you have the solution ?

**Reply by daschr** [reply]
How do you build the project? Seems like some headers are outdated. 
Using platformio (`pio run` and `pio run -t upload`) should work.

**Reply by Zigouille22** [reply]
i use Arduino IDE and i have this same "CAMERA_GRAB_LATEST' was not declared in this scope"  on the main branch
i will try with platformio

**Reply by Zigouille22** [reply]
collect2.exe: error: ld returned 1 exit status
*** [.pio\build\esp32dev\firmware.elf] Error 1
now i have this error when i compil the file
and there is this message in the platformio file:
================================================================
; =                                                              =
; =    DO NOT USE PLATFORMIO FOR THIS PROJECT                    =
; =                                                              =
; = 14 March 2022 : When built with Platformio this project      =
; =   does not run properly; any streams you start will fail     =
; =   with 'ESP_ERR_HTTPD_RESP_SEND' in the serial console and   =
; =   ERR_INVALID_CHUNK_ENCODING 200 (OK) in the browser console.=
; =                                                              =
; = This is a difficult issue, and currently unresolved.         =
; = See:                                                         =
; = https://github.com/easytarget/esp32-cam-webserver/issues/218 =
; =                                                              =
; = The focus of thei project is to show a expanded example      =
; = using the official Arduino IDE. PlatformIO is, and has       =
; = always been, a nice-to-have; so I will be releasing 4.x      =
; = without platformio support. See the above github issue if    =
; = you are able to help fixing this.                            =
; =                                                              =
; ================================================================
;

**Reply by pierroo22** [reply]
@daschr, Hello i clone the multi-stream branch but It appears that only one person can access the stream at the same time. Is there anything to change in a config to fix this ?

**Comment by easytarget** [comment]
Sorry; I am not doing any further development on this project. it is old code and somewhat obsolete. I am currently working on retiring it; I will try to link to a more actively updated fork when I do.
